### DISCUSSION TITLE
Dropping group of cells to improve pseudotime projection

### DISCUSSION THREAD

**Discussion by mitchTwoTimes** [theislab/scvelo#903] [General]
Hi,

Wondering what the maintainer's opinion is on removing cells to improve the visualization of RNA velocity. I'm analyzing an experiment with ipsc-derived cells over the course of 4 weeks. There is a sub-population of cells that doesn't differentiate, which seems to significantly skew the velocity pseudotime. 
![image](https://user-images.githubusercontent.com/92815743/180079496-2547b813-18e1-434c-99c7-3d1824028705.png)

When I remove these ~300 cells and re-run `scVelo`, my dynamic range improves and I can make better inferences on start/end points.
![image](https://user-images.githubusercontent.com/92815743/180079894-ea852bc1-074f-4b1a-a510-e4221745957a.png)

Is this an appropriate approach? FWIW I'm using Bioconductor Package `velociraptor`, version 1.2.0.

Thanks!

**Comment by WeilerP** [comment]
@mitchTwoTimes, sorry, I'm a bit confused: RNA velocity analysis is only applicable if the developmental process is comparable to the time scale of splicing dynamics. This does not seem to be the case here if you are studying a process over four weeks. Or am I missing something.

Either way, I would suggest not to overinterpret low dimensional visualizations. To quantify if RNA velocity is applicable (_i.e._ if the above mentioned criteria is met and to see if model assumptions hold), you should check the phase portraits (you can start e.g. with the genes that are assigned the highest likelihood) and make sure that you see the required almond shape, and that cell types are ordered along the trajectory according to what you expect. This can help you understand why these cells are causing problems. It might be that they form an outlier cluster in the phase portraits and thereby skew the overall fit.

**Reply by mitchTwoTimes** [reply]
@WeilerP Thanks for the prompt reply and the suggestions. Are you saying I can only look at intra-day changes with RNA velocity? The tool actually maps to the time course data quite well, but I understand your caution against over-interpretation.
![image](https://user-images.githubusercontent.com/92815743/180086064-28cb7c6c-f99d-4868-b190-b3c55ace7904.png)

**Reply by WeilerP** [reply]
> Are you saying I can only look at intra-day changes with RNA velocity?

Yes, I think that'd be reasonable here given the fact that your experimental time points are pretty far apart from one another.

> The tool actually maps to the time course data quite well, but I understand your caution against over-interpretation.

Based on the big gaps this seems to be by chance and not because the model assumptions of RNA velocity inference hold (and thereby RNA velocity being inferred reliably).

**Reply by mitchTwoTimes** [reply]
Hi,

Just wanted to follow up on this. I subset to a single day to look at intra-time-point changes, as you suggested. Our outlier cluster makes sense biologically (undifferentiated iPSC), but it seems to really skew pseudotime. The cluster is 40 cells - would it make sense to drop these now?

<img width="719" alt="image" src="https://user-images.githubusercontent.com/92815743/181647525-b673dc9f-0c03-493a-be5b-98b3ec7bc999.png">


Thanks again!

**Reply by WeilerP** [reply]
@mitchTwoTimes, as I mentioned in my first message: You need to look at the phase portraits to understand if RNA velocity is applicable. This can also help you understand if you should exclude this cluster.
I'd also try to understand why these cells form a distinct cluster really separate from the other cells. Something along the lines of if they should have been filtered out by quality control or not, etc.

**Reply by mitchTwoTimes** [reply]
`velociraptor` doesn't have a function to make phase portraits so this is my best attempt to recreate these plots with the `spliced` and `unspliced` assays. I took the top 6 r2 velocity genes from the scVelo output and colored points by cluster and velocity pseudotime. We've already assured ourselves that the small, "pseudotime 0", cluster is legitimate biology and not a poor-QC cluster.
```
> head(velo_genes[order(velo_genes$velocity_r2, decreasing = TRUE), ], 6)
DataFrame with 6 rows and 3 columns
                velocity_gamma velocity_r2 velocity_genes
                     <numeric>   <numeric>      <logical>
ENSG00000105825      0.0229238    0.994819           TRUE
ENSG00000164692      0.3367712    0.988814           TRUE
ENSG00000106278      2.2351515    0.986200           TRUE
ENSG00000048540      4.8369231    0.909592           TRUE
ENSG00000164741     18.9171028    0.907725           TRUE
ENSG00000196628      2.8650386    0.885981           TRUE

```

![image](https://user-images.githubusercontent.com/92815743/184039715-b0683441-2d77-4b34-b94a-a415eaa82d30.png)

@WeilerP is this what you had in mind? I'm not seeing the almond shape - though I'm not completely confident I did this correctly.

**Reply by mitchTwoTimes** [reply]
@WeilerP would like to circle back to this, please. Can we assume the assumptions are met?

**Reply by WeilerP** [reply]
@mitchTwoTimes, sorry for the delay. You need to visualize the imputed data, _i.e._, the first order moments to check if your phase portraits include enough signal to infer RNA velocity.
